general/pages/connect/github.vue

```diff
@@ -0,0 +1,128 @@
+          />
+        </svg>
+        <h1>Connecting</h1>
+        <p class="muted">{{ statusMessage }}</p>
+      </section>
+    </article>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted } from 'vue';
+import { useRoute, useRouter } from 'vue-router';
+import { useAuthStore, type UserData } from '~/stores/auth';
+import { useToastStore } from '~/stores/toast';
+import { useRuntimeConfig } from '#app';
+
+definePageMeta({
+  layout: 'empty'
+});
+
+const route = useRoute();
+const router = useRouter();
+const authStore = useAuthStore();
+const toastStore = useToastStore();
+const runtimeConfig = useRuntimeConfig();
+const configStore = useConfigurationStore();
+
+const statusMessage = ref('Processing GitHub authentication...');
+
+const baseURL = () => {
+  return typeof window !== 'undefined'
+    ? window.location.origin + `${runtimeConfig.public.strapiBasePath}`
+    : `${runtimeConfig.public.strapiBaseUrl}`;
+};
+
+interface AuthResponse {
+  jwt: string;
+  user: UserData;
+}
+
+onMounted(async () => {
+  if (configStore.config.authEnabled !== 'true') {
+    handleError('Authentication is not enabled');
+    return;
+  }
+
+  const accessToken = route.query.access_token as string;
+
+  if (!accessToken) {
+    handleError('No access token provided');
+    return;
+  }
+
+  try {
+    const response = await $fetch<AuthResponse>(
+      `${baseURL()}/api/auth/github/callback`,
+      {
+        method: 'GET',
+        params: { access_token: accessToken }
+      }
+    );
+
+    if (response.jwt && response.user) {
+      authStore.setJwt(response.jwt);
+      authStore.setUserData(response.user);
+      toastStore.addToast('Successfully logged in with GitHub.');
+      statusMessage.value = 'Authentication successful. Redirecting...';
+      setTimeout(() => router.push('/ontology'), 2000);
+    } else {
+      throw new Error('Invalid response from server');
+    }
+  } catch (error: any) {
+    handleError(error.message || 'Failed to authenticate with GitHub');
+  }
+});
+
+const handleError = (message: string) => {
+  console.error('GitHub auth error:', message);
+  statusMessage.value = 'Authentication failed. Redirecting...';
+  toastStore.addToast(`GitHub authentication failed!`, 'error');
+  authStore.clear();
+  setTimeout(() => router.push('/'), 2000);
```

Comment:
After a failed login, the user should return to the exit page instead of the [/](https://spec.edmcouncil.org/esg/)